nucypher/crypto/utils.py

```diff
@@ -40,10 +45,185 @@ def construct_policy_id(label: bytes, stamp: bytes) -> bytes:
+def canonical_address_from_umbral_key(public_key: UmbralPublicKey) -> bytes:
+    pubkey_raw_bytes = get_coordinates_as_bytes(public_key)
+    eth_pubkey = EthKeyAPI.PublicKey(pubkey_raw_bytes)
     canonical_address = eth_pubkey.to_canonical_address()
     return canonical_address
+
+
+def get_coordinates_as_bytes(point: Union[Point, UmbralPublicKey, SignatureStamp],
+                             x_coord=True,
+                             y_coord=True) -> bytes:
+    if isinstance(point, SignatureStamp):
+        point = point.as_umbral_pubkey()
+
+    coordinates_as_bytes = point.to_bytes(is_compressed=False)[1:]
+    middle = len(coordinates_as_bytes)//2
+    if x_coord and y_coord:
+        return coordinates_as_bytes
+    elif x_coord:
+        return coordinates_as_bytes[:middle]
+    elif y_coord:
+        return coordinates_as_bytes[middle:]
+    else:
+        raise ValueError("At least one coordinate must be set")
+
+
+SYSTEM_RAND = SystemRandom()
+
+
+def secure_random(num_bytes: int) -> bytes:
+    """
+    Returns an amount `num_bytes` of data from the OS's random device.
+    If a randomness source isn't found, returns a `NotImplementedError`.
+    In this case, a secure random source most likely doesn't exist and
+    randomness will have to found elsewhere.
+
+    :param num_bytes: Number of bytes to return.
+
+    :return: bytes
+    """
+    # TODO: Should we just use os.urandom or avoid the import w/ this?
+    return SYSTEM_RAND.getrandbits(num_bytes * 8).to_bytes(num_bytes, byteorder='big')
+
+
+def secure_random_range(min: int, max: int) -> int:
+    """
+    Returns a number from a secure random source betwee the range of
+    `min` and `max` - 1.
+
+    :param min: Minimum number in the range
+    :param max: Maximum number in the range
+
+    :return: int
+    """
+    return SYSTEM_RAND.randrange(min, max)
+
+
+def keccak_digest(*messages: bytes) -> bytes:
+    """
+    Accepts an iterable containing bytes and digests it returning a
+    Keccak digest of 32 bytes (keccak_256).
+
+    Although we use SHA256 in many cases, we keep keccak handy in order
+    to provide compatibility with the Ethereum blockchain.
+
+    :param bytes: Data to hash
+
+    :rtype: bytes
+    :return: bytestring of digested data
+    """
+    _hash = sha3.keccak_256()
+    for message in messages:
+        _hash.update(bytes(message))
+    digest = _hash.digest()
+    return digest
+
+
+def sha256_digest(*messages: bytes) -> bytes:
+    """
+    Accepts an iterable containing bytes and digests it returning a
+    SHA256 digest of 32 bytes
+
+    :param bytes: Data to hash
+
+    :rtype: bytes
+    :return: bytestring of digested data
+    """
+    _hash_ctx = hashes.Hash(hashes.SHA256(), backend=backend)
+    for message in messages:
+        _hash_ctx.update(bytes(message))
+    digest = _hash_ctx.finalize()
+    return digest
+
+
+def ecdsa_sign(message: bytes,
```

Comment:
Note: it's being removed in #2612. If I'm not mistaken, it does the same as `umbral`'s signing/verifying functions.